modules/graphql_content/src/Plugin/Deriver/EntityQueryDeriver.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\graphql_content\Plugin\Deriver;
+
+use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\Core\Entity\ContentEntityTypeInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Plugin\Discovery\ContainerDeriverInterface;
+use Drupal\Core\TypedData\TypedDataManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Create GraphQL entityQuery fields based on available Drupal entity types.
+ */
+class EntityQueryDeriver extends DeriverBase implements ContainerDeriverInterface {
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The typed data manager service.
+   *
+   * @var \Drupal\Core\TypedData\TypedDataManager
+   */
+  protected $typedDataManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, $basePluginId) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('typed_data_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, TypedDataManager $typedDataManager) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->typedDataManager = $typedDataManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($basePluginDefinition) {
+    foreach ($this->entityTypeManager->getDefinitions() as $id => $type) {
+      if ($type instanceof ContentEntityTypeInterface) {
+        $derivative = [
+          'name' => graphql_core_propcase($id) . 'Query',
+          'type' => graphql_core_camelcase($id),
```

Comment:
If `graphql_core` is not enabled, or certain entity types are not included in the schema (#86), this will break. We would have to set the type to `Entity`, as in #90.